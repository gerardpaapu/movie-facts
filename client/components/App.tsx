import { Routes, Route, Link } from 'react-router-dom'
import MovieList from './MovieList'
import Movie from './Movie'
import Search from './Search'
import Home from './Home'
import Category from './Category'
import CategoryList from './CategoryList'

function App() {
  return (
    <>
      <header className="header">
        <h1>Movie Facts</h1>
      </header>
      <nav>
        <ul>
          <li>
            <Link to="/movie">All movies</Link>
          </li>

          <li>
            <Link to="/category">All categories</Link>
          </li>
          <li>
            <Link to="/search">Search</Link>
          </li>
        </ul>
      </nav>
      <section className="main">
        <Routes>
          <Route index element={<Home />} />
          <Route path="/movie">
            <Route index element={<MovieList />} />
            <Route path=":id" element={<Movie />} />
          </Route>
          <Route path="/category">
            <Route index element={<CategoryList />} />
            <Route path=":id" element={<Category />} />
          </Route>
          <Route path="/search" element={<Search />} />
        </Routes>
      </section>
    </>
  )
}

export default App
