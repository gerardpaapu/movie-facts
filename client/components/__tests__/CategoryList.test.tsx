/** @jest-environment jsdom */
import nock from 'nock'
import App from '../App'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { initialiseStore } from '../../store'

import { render, waitFor } from '@testing-library/react'

import '@testing-library/jest-dom'

describe('<Category />', () => {
  it('Shows information about the category', async () => {
    const scope = nock('http://localhost')
      .get('/api/v1/categories')
      .reply(200, [
        {
          id: 1,
          name: 'Action/Adventure',
        },
        {
          id: 2,
          name: 'Sci-Fi',
        },
        {
          id: 3,
          name: 'Drama',
        },
        {
          id: 4,
          name: 'Thriller',
        },
        {
          id: 5,
          name: 'Comedy',
        },
        {
          id: 6,
          name: 'Fantasy',
        },
        {
          id: 7,
          name: 'History',
        },
        {
          id: 8,
          name: 'Western',
        },
        {
          id: 9,
          name: 'Musical',
        },
      ])

    const { container } = render(
      <Router initialEntries={['/category']}>
        <Provider store={initialiseStore()}>
          <App />
        </Provider>
      </Router>
    )

    await waitFor(() => expect(scope.isDone()).toBe(true))
    expect(container).toMatchSnapshot()
  })
})
