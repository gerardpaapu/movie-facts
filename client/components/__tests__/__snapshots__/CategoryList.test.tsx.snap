// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> Handles errors 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Movie Facts
    </h1>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="/movie"
        >
          All movies
        </a>
      </li>
      <li>
        <a
          href="/category"
        >
          All categories
        </a>
      </li>
      <li>
        <a
          href="/search"
        >
          Search
        </a>
      </li>
    </ul>
  </nav>
  <section
    class="main"
  >
    <p>
      Error! 
      Internal Server Error
    </p>
  </section>
</div>
`;

exports[`<Category /> Shows information about the category 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Movie Facts
    </h1>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="/movie"
        >
          All movies
        </a>
      </li>
      <li>
        <a
          href="/category"
        >
          All categories
        </a>
      </li>
      <li>
        <a
          href="/search"
        >
          Search
        </a>
      </li>
    </ul>
  </nav>
  <section
    class="main"
  >
    <div>
      <h2>
        All categories
      </h2>
      <ul>
        <li>
          <a
            href="/category/1"
          >
            Action/Adventure
          </a>
        </li>
        <li>
          <a
            href="/category/2"
          >
            Sci-Fi
          </a>
        </li>
        <li>
          <a
            href="/category/3"
          >
            Drama
          </a>
        </li>
        <li>
          <a
            href="/category/4"
          >
            Thriller
          </a>
        </li>
        <li>
          <a
            href="/category/5"
          >
            Comedy
          </a>
        </li>
        <li>
          <a
            href="/category/6"
          >
            Fantasy
          </a>
        </li>
        <li>
          <a
            href="/category/7"
          >
            History
          </a>
        </li>
        <li>
          <a
            href="/category/8"
          >
            Western
          </a>
        </li>
        <li>
          <a
            href="/category/9"
          >
            Musical
          </a>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
